hazelcast/src/test/java/com/hazelcast/cache/impl/AbstracRecordStoreTest.java

```diff
@@ -0,0 +1,106 @@
+package com.hazelcast.cache.impl;
+
+import com.hazelcast.cache.CacheFromDifferentNodesTest;
+import com.hazelcast.cache.CacheTestSupport;
+import com.hazelcast.cache.HazelcastCacheManager;
+import com.hazelcast.config.CacheConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.instance.TestUtil;
+import com.hazelcast.test.AssertTask;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import org.junit.Test;
+
+import javax.cache.Cache;
+import javax.cache.configuration.FactoryBuilder;
+import javax.cache.configuration.MutableCacheEntryListenerConfiguration;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class AbstracRecordStoreTest extends CacheTestSupport {
+
+    TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory();
+
+    @Test
+    public void testRemovingSameEntryTwiceShouldTriggerEntryListenerOnlyOnce() {
+        String cacheName = randomString();
+
+        CacheConfig<Integer, String> config = createCacheConfig();
+        final CacheFromDifferentNodesTest.SimpleEntryListener<Integer, String> listener =
+                new CacheFromDifferentNodesTest.SimpleEntryListener<Integer, String>();
+        MutableCacheEntryListenerConfiguration<Integer, String> listenerConfiguration =
+                new MutableCacheEntryListenerConfiguration<Integer, String>(
+                        FactoryBuilder.factoryOf(listener), null, true, true);
+
+        config.addCacheEntryListenerConfiguration(listenerConfiguration);
+
+        Cache<Integer, String> cache = cacheManager.createCache(cacheName, config);
+        assertNotNull(cache);
+
+        Integer key = 1;
+        String value = "value";
+        cache.put(key, value);
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run()
+                    throws Exception {
+                assertEquals(1, listener.created.get());
+            }
+        });
+
+        cache.removeAll();
+        cache.removeAll();
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run()
+                    throws Exception {
+                assertEquals(1, listener.removed.get());
+            }
+        });
+    }
+
+    //https://github.com/hazelcast/hazelcast/issues/6618
+    @Test
+    public void testBatchEventMapShouldBeCleanedAfterRemoveAll() {
```

Comment:
`CacheRecordStoreTest` cannot access `RecordStore`'s `batchEvent` map as they are in different packages. That is why I created this new class in the first place. What should I do?